File: ruby/bench.rb
module Bench
    def execute(target, count)
        load "#{File.dirname(__FILE__)}/script/#{target}.rb"
        target_module = eval target.capitalize

        start = Time.now
        count.times {
            target_module.main()
        }

        return Time.now - start        
    end

    module_function :execute

    count = ARGV[0].to_i
    blank = Bench.execute('blank', count)
    ARGV[1..ARGV.size - 1].each do |target|
        time = Bench.execute(target, count) - blank
        puts "# #{target}"
        puts "## #{time}"
    end
end

File: node/bench.js
function execute(target, count) {
  var targetModule = require('./script/' + target)
  var func = targetModule.main
  var start = process.hrtime()
  for(var i = 0; i < count; i++) {
    func()
  }
  var end = process.hrtime(start)
  return end[1] / 1e9 + end[0]
}

var count = parseInt(process.argv[2])
var targetList = process.argv.slice(3)
var blank = execute('blank', count)
for(var i in targetList) {
  var time = execute(targetList[i], count) - blank
  console.log('# ' + targetList[i])
  console.log('## ' + time)
}
